homework6/test/test2.py

```diff
@@ -0,0 +1,81 @@
+import datetime
+
+import homework6.oop_2 as hw1
+import pytest
+
+
+def test_teacher_created():
+    teacher = hw1.Teacher("Daniil", "Shadrin")
+    assert (teacher.last_name, teacher.first_name) == ("Shadrin", "Daniil")
+
+
+def test_student_created():
+    student = hw1.Student("Martin", "Septim")
+    assert (student.last_name, student.first_name) == ("Septim", "Martin")
+
+
+def test_homework_created_without_teacher_object():
+    hw = hw1.Teacher.create_homework("do something", 10)
+    assert (hw.deadline, hw.text) == (datetime.timedelta(days=10), "do something")
+
+
+def test_homework_late():
+    hw = hw1.Teacher.create_homework("late", 0)
+    with pytest.raises(hw1.DeadlineError):
+        student = hw1.Student("Jon", "Snow")
```

Comment:
actually, 25 line does not raise the error, but 26 does. move this line out of the check scope